src/main/java/io/luna/net/msg/out/PlayerUpdateMessageWriter.java

```diff
@@ -35,6 +43,15 @@ public ByteMessage write(Player player) {
             msg.startBitAccess();
 
             handleMovement(player, msg);
+
+            Music m = Music.getSongForReqion(player.getPosition().getRegionPosition().getId());
+            if (m!=null) {
+                m.playSong(m, player);
+            } else {
+                LOGGER.info("No music for region: " + player.getPosition().getRegionPosition().getId());
+                Music.stopMusic(player);
```

Comment:
Should create a new event for this `RegionIdChangedEvent` so we only have to check for updates when the region id actually changes. Also that way you don't have to worry about threading issues (bc player updating is executed using a thread pool)